server/auth/manager.py

```
@@ -0,0 +1,83 @@
+                    else:
+                        mylog("verbose", ["[auth.manager] User not found in LDAP, but local fallback is disabled."])
+                        return ldap_result
+
+                # LDAP returned failure without exception, which means invalid credentials
+                # We should NOT fallback if the LDAP server explicitly rejected them
+                if not disable_local:
+                    mylog("warning", ["[auth.manager] LDAP explicitly rejected credentials, no fallback to local provider"])
+                else:
+                    mylog("verbose", [f"[auth.manager] Authentication failed for user '{_sanitize_for_log(username)}' via ldap"])
+                return ldap_result
+            except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)